### PR TITLE
Improve test suite performance using Test Dataset

### PR DESCRIPTION
**Pull request by balajisivaraman**
The objective of this PR is to address #655, as much as possible.

I have opened this as a draft PR for the time being so that I can continue adding commits to it for review.

I will aim to complete the following activities in this PR, as individual commits:

- [x] Documents Delete
- [x] Settings
- [x] Search
- [x] Search Settings
- [x] Settings Stop Words
- [x] Settings Ranking Rules

### CODE REVIEW

**Comment by MarinPostma** [comment]
Hello @balajisivaraman ! I hope you haven't started to work the search tests, I am doing them, because I need it for #735. Cheers :)

**Comment by balajisivaraman** [comment]
@MarinPostma, Sure, thanks for letting me know. Does this mean you'll also change it to use Serde URL Params or shall I do it?

**Comment by MarinPostma** [comment]
I made the change to serde-url-params for this one, just ignore this file altogether

**Comment by MarinPostma** [comment]
Hello @balajisivaraman ! are you still working on this?

**Comment by balajisivaraman** [comment]
@MarinPostma, I didn't get to work on it over the past 2 weekends. I could work on it this weekend. However, if there are pressing needs, I wouldn't mind this PR being merged also.

**Comment by MarinPostma** [comment]
Don't worry, there is no rush 😁 take the time you need!

**Comment by balajisivaraman** [comment]
@MarinPostma, I still see that `populate_movies` is used in a number of places in the `search` test suite. If you want, I can go ahead and change them. Those are the only ones remaining now.

**Comment by MarinPostma** [comment]
That would be great indeed! Thank you for your work 😉

**Comment by balajisivaraman** [comment]
I was working on the `search` tests, and it appears to me that `search_with_attribute_to_highlight_title_tagline` and `search_with_attribute_to_highlight_title_overview` are testing the same thing as `search_with_attribute_to_highlight_1`. I have removed what I felt were duplicates, but please let me know if I have to restore them using the test dataset.

Other than that, this is now complete and ready for review. :-)

File: meilisearch-http/tests/search_settings.rs [hunk 1]
```diff
@@ -228,98 +176,72 @@ async fn search_with_settings_synonyms() {
...
       ],
       "stopWords": null,
       "synonyms": {
-        "avangers": [
-          "Captain America",
-          "Iron Man"
-        ]
+          "exarcitation": [
+              "exercitation"
+          ]
```
Comment (MarinPostma) [inline meilisearch-http/tests/search_settings.rs:207]: uhm, I think this synonym is too close to the original word, and actually is caught by the typo system. Can you try with a word that is unrelated to make sure the correct documents are picked?

Comment (balajisivaraman) [reply meilisearch-http/tests/search_settings.rs:207]: Oh, my bad. I can understand now. Will update it with a different word. 👍

**Comment by balajisivaraman** [comment]
@MarinPostma, Addressed both comments, please check. :+1:

**Comment by balajisivaraman** [comment]
Hey @Kerollmops, it's done. 😄
